b2share/modules/deposit/api.py

```diff
@@ -224,6 +305,26 @@ def submit(self, commit=True):
         if commit:
             self.commit()
 
+    @classmethod
+    def delete(self, pid_value):
+        """Delete a deposit."""
+        record_pid = RecordUUIDProvider.get(pid_value).pid
+        version_child = PIDVersioning(child=record_pid)
+        version_master = PIDVersioning(parent=version_child.parent)
+        if version_master is not None \
+                and version_master.draft_child.pid_value == pid_value:
+            pid = PersistentIdentifier.query.\
+                filter_by(pid_value=pid_value, pid_type='b2dep').one_or_none()
+            if not pid:
+                raise ValueError()
+            version_master.remove_draft_child()
+            pid.delete()
+            # hard delete the record pid
+            db.session.delete(record_pid)
+            if version_child.parent.status == PIDStatus.RESERVED:
+                db.session.delete(version_child.parent)
+            db.session.commit()
```

Comment:
deposit should also be removed from the ES index.

Reply:
to be tested.